### PR TITLE
Nn matrix geometry try2

### PR DESCRIPTION
**Pull request by cvvergara**
Removing warnings from merged branch nn-matrix-geometry
(see #4435 )

### CODE REVIEW

**Comment by kevinkreiser** [comment]
can you keep the headers private? id rather not expand the libraries public api

**Comment by cvvergara** [comment]
I get you, I will start all over and leave the header problems to another PR

**Comment by cvvergara** [comment]
hmm, I will not move them to valhalla/tyr first to see if it works

**Comment by cvvergara** [comment]
@kevinkreiser  tyr headers are now in src/tyr directory

File: src/tyr/route_serializer.cc [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 #include "midgard/encoded.h"
 #include "midgard/util.h"
-#include "route_serializer_osrm.cc"
+#include "route_serializer_osrm.h"
```
Comment (nilsnolde) [inline src/tyr/route_serializer.cc:8]: yeah, so much nicer to look at:)

File: src/tyr/route_summary_cache.h [hunk 2]
```diff
@@ -1,3 +1,5 @@
+#ifndef VALHALLA_TYR_ROUTE_SUMMARY_CACHE_H_
+#define VALHALLA_TYR_ROUTE_SUMMARY_CACHE_H_
 #pragma once
```
Comment (nilsnolde) [inline src/tyr/route_summary_cache.h:3]: shouldn't `pragma once` be enough here? I thought it's properly supported by most toolchains by now?

File: valhalla/tyr/serializer_constants.h [hunk 3]
```diff
@@ -1,3 +1,6 @@
+#ifndef VALHALLA_TYR_ROUTE_SERIALIZER_CONSTANTS_
+#define VALHALLA_TYR_ROUTE_SERIALIZER_CONSTANTS_
+#pragma once
```
Comment (nilsnolde) [inline valhalla/tyr/serializer_constants.h:3]: same here

**Review by nilsnolde — COMMENTED** [review]
I guess you hardly changed anything in `route_serializer` & `route_summary_cache` other than splitting them?

File: src/tyr/route_summary_cache.h (hunk 2 above)
Comment (cvvergara) [reply src/tyr/route_summary_cache.h:3]: `pragma once` is not standard

**Review by nilsnolde — DISMISSED** [review]
LGTM

File: valhalla/tyr/serializer_constants.h (hunk 3 above)
Comment (nilsnolde) [reply valhalla/tyr/serializer_constants.h:3]: let's just remove those `pragma once` then

**Review by nilsnolde — APPROVED** [review]

**Pull request merged**